src/components/client/search-result-ui.tsx

```diff
@@ -0,0 +1,106 @@
+import Image from "next/image";
+import React from "react";
+
+import { SearchQuery, SearchResults } from "./search-modal";
+
+type SearchResultProps = {
+  searchResults: SearchResults | undefined;
+  isPending: boolean;
+  searchQuery: SearchQuery | null;
+  showMoreResults: () => void;
+  limit: number;
+};
+
+function SearchResult({
+  searchResults,
+  searchQuery,
+  isPending,
+  showMoreResults,
+  limit,
+}: SearchResultProps) {
+  if (isPending || !searchQuery) return null;
+
+  if (
+    searchResults?.searchResults.length === 0 ||
+    searchResults?.totalSearchResults === 0
+  ) {
+    return (
+      <p className="text-center">
+        Oops! we could not find what you are looking for. You can try searching
+        for a similar keyword
+      </p>
+    );
+  }
+
+  function sortData() {
+    if (!searchResults) return;
+    let searchData: SearchResults["searchResults"];
+    switch (searchQuery?.path) {
+      case "bitcoin-dev":
+        searchData = searchResults?.searchResults.filter((result) => {
+          return result.path.includes("bitcoin-dev");
+        });
+        break;
+      case "lightning-dev":
+        searchData = searchResults?.searchResults.filter((result) => {
+          return result.path.includes("lightning-dev");
+        });
+        break;
+      default:
+        searchData = searchResults?.searchResults;
+        break;
+    }
+
+    return searchData.slice(0, limit);
+  }
+
+  const showMore = searchResults && limit >= searchResults?.totalSearchResults;
+
+  return (
+    <div className="w-full">
+      <ul className="mt-0">
+        {sortData()?.map((result, index) => {
```

Comment:
we are sorting the data here, but we are also running the query in search-modal.tsx (It runs the query all over again, the subsequent runs are faster though I'm assuming OS disk caching)

Reply:
We run the query just one time in search modal. Disk caching is what happens here and that's why the sort is faster i believe